### PR TITLE
CLI Tricks for GitHub

### PR DESCRIPTION
**Pull request by emmasofiah2o**
- Add blog post

To-do:
- [ ]  Add og-image
- [ ]  Add illustration

### CODE REVIEW

**Comment by emmasofiah2o** [comment]
@marcoow Bart had this idea about a TIL post a while back. I have helped him a little structuring the post and this is what came out. It's short, but I think it could work. What do you say?

File: src/posts/2023-05-15-cli-tricks.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+title: "CLI Tricks for GitHub"
```
Comment (BobrImperator) [inline src/posts/2023-05-15-cli-tricks.md]:
```suggestion
title: "CLI Tricks for Git with FZF"
```

Github is a service that hosts Git repositories.
Some people could be using e.g. Gitlab or have a self-hosted solution to host their repositories but still use `Git` as their version control tool.

File: src/posts/2023-05-15-cli-tricks.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+authorHandle: BobrImperator
+tags: [misc]
+bio: "Bartłomiej Dudzik"
+description:
+  "Bartłomiej Dudzik shows a trick to checkout a GitHub branch faster in the
+  CLI."
```
Comment (BobrImperator) [inline src/posts/2023-05-15-cli-tricks.md]:
```suggestion
  "Bartłomiej Dudzik shows a trick to checkout a Git branch faster in the
  CLI."
```

File: src/posts/2023-05-15-cli-tricks.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+og:
+  image:
+tagline: |
+  <p>Are you looking to be more efficient with Git while wokring with the CLI? There is an easy way to make checking out branches a one-step process.</p>
```
Comment (BobrImperator) [inline src/posts/2023-05-15-cli-tricks.md]:
```suggestion
  <p>Are you looking to be more efficient with Git while working with the CLI? There is an easy way to make checking out branches a one-step process.</p>
```

File: src/posts/2023-05-15-cli-tricks.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+image: ""
+imageAlt: ""
+---
+
+Does your current process to checkout a GitHub branch look something like this?
```
Comment (BobrImperator) [inline src/posts/2023-05-15-cli-tricks.md]:
```suggestion
Does your current process to checkout a Git branch look something like this?
```

File: src/posts/2023-05-15-cli-tricks.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+- `git branch` or `git branch --list`
+- outputs a non-interactive list of branches
+- Double-click an entry
+- do `CTRL-C` or right click and copy
+- close the list
+- paste the entry to the terminal
+- prepend entry with `git checkout`
+
+Going through these many steps decreases productivity and makes working with
+GitHub a real struggle. The good news is, there is a way to work around this and
```
Comment (BobrImperator) [inline src/posts/2023-05-15-cli-tricks.md]:
```suggestion
Git a real struggle. The good news is, there is a way to work around this and
```

File: src/posts/2023-05-15-cli-tricks.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+make the whole process much leaner!
+
+Using FZF, the seven steps listed above become a one-liner:
```
Comment (BobrImperator) [inline src/posts/2023-05-15-cli-tricks.md]: FZF is a tool that is usually not included on new systems so people will have to install it themselves
```suggestion
Using [FZF](https://github.com/junegunn/fzf), the seven steps listed above become a one-liner:
```

File: src/posts/2023-05-15-cli-tricks.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+`~/.bashrc`
+
+`alias gb="git branch | fzf | xargs git checkout"`
+
+You should now be able to use this with a `gb` shorthand, making your experience
+smoother. Happy coding!
+
+If you are looking for a guide on using Fork to manage your Git actions, check
+out this blog post by Chris Manson:
```
Comment (BobrImperator) [inline src/posts/2023-05-15-cli-tricks.md]:
```suggestion
If you are looking for a guide on using Git with visual tools like Fork, check out this blog post by Chris Manson:
```

File: src/posts/2023-05-15-cli-tricks.md (hunk 3 above)
Comment (marcoow) [reply src/posts/2023-05-15-cli-tricks.md]: Checking out branches is a one step process?

```bash
git checkout <branch>
```

**Review by marcoow — COMMENTED** [review]
I'm not sure this is a blog post tbh 😬 It's definitely another signal that having the TIL would be cool though!

We can make it into a blog post if we want to but I think we'd have to explain a bit better what this is about. I didn't get it the first time I read it (see my comment)

**Comment by emmasofiah2o** [comment]
> I'm not sure this is a blog post tbh 😬 It's definitely another signal that having the TIL would be cool though!
> 
> We can make it into a blog post if we want to but I think we'd have to explain a bit better what this is about. I didn't get it the first time I read it (see my comment)

I see. Would you recommend turning it into a blog post or rather using it in another way (for example LinkedIn article, but that way we only post on one platform)? I am trying to understand if it makes sense to invest more time into this since I would also need Bart's support.

**Comment by BobrImperator** [comment]
It really feels like the tagline is the only issue here.
IMO checking out a branch is not a one-step process if you don't know the name of it or at least the ticket number it relates to (if it has one). I'd often forget the ticket number, or knew only a part of the name that the usual autocompletion wouldn't pick up easily. While the script from the post gives me an ability to have a searchable list where i just need to select an item I'm interested in.

If we were to say:
```
Are you looking to browse your Git branches locally in more efficient way? There is an easy way to make checking out branches a one-step process.
```
Then it feels to me like part of the issue goes away.

**Comment by marcoow** [comment]
we can grow this into a blog post or use differently

**Comment by emmasofiah2o** [comment]
@BobrImperator I'd say let's use this as a LinkedIn article. That way we don't need to have a TIL section on the website for these short blog posts and we can have a new "series" for LinkedIn to replace carousels :)

I'll work through Marco's suggestions later today and see if I can make the text clearer.

**Comment by emmasofiah2o** [comment]
Posted on LinkedIn as article ✅

**Pull request closed without merging**